From 41579f76f82137f39de38a160abad18579cbb9c4 Mon Sep 17 00:00:00 2001
From: Marc Fournier <marc.fournier@camptocamp.com>
Date: Thu, 25 Feb 2016 11:22:18 +0100
Subject: [PATCH] modbus: avoid enabling libmodbus's debug flag by default

Having this enabled makes collectd's stdout and stderr very noisy. From
now on, only activate this flag when collectd is built with
--enable-debug.

Thanks to Eric Sandeen for mentioning this problem on IRC.
---
 src/modbus.c | 4 ++++
 1 file changed, 4 insertions(+)

diff --git a/src/modbus.c b/src/modbus.c
index 97b98b7..c04b308 100644
--- a/src/modbus.c
+++ b/src/modbus.c
@@ -310,7 +310,9 @@ static int mb_init_connection (mb_host_t *host) /* {{{ */
   if (host == NULL)
     return (EINVAL);
 
+#if COLLECT_DEBUG
   modbus_set_debug (&host->connection, 1);
+#endif
 
   /* We'll do the error handling ourselves. */
   modbus_set_error_handling (&host->connection, NOP_ON_ERROR);
@@ -390,7 +392,9 @@ static int mb_init_connection (mb_host_t *host) /* {{{ */
     }
   }
 
+#if COLLECT_DEBUG
   modbus_set_debug (host->connection, 1);
+#endif
 
   /* We'll do the error handling ourselves. */
   modbus_set_error_recovery (host->connection, 0);
-- 
2.5.5

